lib/OpenQA/Command.pm

```diff
@@ -81,14 +82,19 @@ sub parse_headers {
 }
 
 sub parse_params {
-    my ($self, @args) = @_;
+    my ($self, $args, $param_file) = @_;
 
     my %params;
-    for my $arg (@args) {
+    for my $arg (@{$args}) {
         next unless $arg =~ /^([[:alnum:]_\[\]\.]+)=(.*)$/s;
         push @{$params{$1}}, $2;
     }
 
+    for my $arg (@{$param_file}) {
+        next unless $arg =~ /^([[:alnum:]_\[\]\.]+)=(.*)$/s;
```

Comment:
The regex is long enough that it should be factored out into a global for reuse.
```perl
$PARAM_RE = qr/^([[:alnum:]_\[\]\.]+)=(.*)$/s;
```
and then here
```perl
next unless $arg =~ $PARAM_RE;
```

Reply:
Done.